components/bt/host/bluedroid/api/include/api/esp_gap_ble_api.h

```diff
@@ -17,2687 +17,3971 @@
+    int min_interval;               /*!< Slave preferred connection minimum interval.
+                                         - Unit: 1.25ms.
+                                         - Range: 0x0006 to 0x0C80 (7.5ms to4 s). */
+    int max_interval;               /*!< Slave preferred connection maximum interval.
+                                         - Unit: 1.25ms.
+                                         - Range: 0x0006 to 0x0C80 (7.5ms to 4s).
+                                         - max_interval shall be equal to or greater than min_interval. */
+    int appearance;                 /*!< External appearance of the device.
+                                         - The value is defined by the Bluetooth SIG and represents the appearance of the device.
+                                         - See `esp_ble_icon_t` for details. 
+                                         - Group define for `ESP_BLE_APPEARANCE_xxx`. */
+    uint16_t manufacturer_len;      /*!< Manufacturer data length in bytes. */
+    uint8_t *p_manufacturer_data;   /*!< Pointer to the manufacturer data.
+                                         - Manufacturer data is typically used to transmit device vendor-specific information. */
+    uint16_t service_data_len;      /*!< Service data length in bytes. */
+    uint8_t *p_service_data;        /*!< Pointer to the service data.
+                                         - The Service Data data type consists of a service UUID with the data associated with that service. */
+    uint16_t service_uuid_len;      /*!< Service UUID list length in bytes. */
+    uint8_t *p_service_uuid;        /*!< Pointer to the service UUID list.
+                                         - The Service UUID data type is used to include a list of Service or Service Class UUIDs.
+                                         - The UUID in the list is 16/32/128-bit Service UUIDs. */
+    uint8_t flag;                   /*!< Advertising flag of discovery mode.
+                                         - Used to specify the discovery mode and function of the device.
+                                         - See `esp_ble_adv_flag_t` for details.
+                                         - Group define for `ESP_BLE_ADV_FLAG_xxx`. */
+} esp_ble_adv_data_t;// #if (BLE_42_FEATURE_SUPPORT == TRUE)
+#endif
 
-/// Ble scan type
+/**
+ * @brief Bluetooth Low Energy (LE) scan type
+ */
 typedef enum {
-    BLE_SCAN_TYPE_PASSIVE   =   0x0,            /*!< Passive scan */
-    BLE_SCAN_TYPE_ACTIVE    =   0x1,            /*!< Active scan */
+    BLE_SCAN_TYPE_PASSIVE   =   0x0, /*!< Passive scan. No scanning Protocol Data Unit (PDU) shall be sent. */
+    BLE_SCAN_TYPE_ACTIVE    =   0x1, /*!< Active scan. Scanning PDU may be sent. */
 } esp_ble_scan_type_t;
 
-/// Ble scan filter type
+/**
+ * @brief Bluetooth Low Energy (LE) scan filter type
```

Comment:
1 If you want to filter device using white list, set the scan filter policy to `BLE_SCAN_FILTER_ALLOW_ONLY_WLST`.
2 If you want to filter device using white list and recevie directed advertising packet where the initiator address is RPA, set the scan filter policy to `BLE_SCAN_FILTER_ALLOW_WLIST_RPA_DIR`.